BUG: Occurs panic

**Describe this problem**
Refer to https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x81dbcd]

goroutine 1 [running]:
github.com/CeresDB/ceresdb-client-go/ceresdb.(*rpcClient).Write(0xc000216600?, {0x9e51e8, 0xc0000300e8}, {0x9[29](https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732#step:8:30)8bd?, 0x7993e5?}, {{0x9249[30](https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732#step:8:31)?, 0x1?}}, {0xc0001926e0, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/!ceres!d!b/ceresdb-client-go@v1.1.0/ceresdb/rpc.go:96 +0x12d
github.com/CeresDB/ceresdb-client-go/ceresdb.(*clientImpl).Write(0xc00000e6c0, {0x9e51e8, 0xc0000300e8}, {{{0x924930, 0x6}}, {0xc00007cc80, 0x2, 0x2}})
	/home/runner/go/pkg/mod/github.com/!ceres!d!b/ceresdb-client-go@v1.1.0/ceresdb/client_impl.go:75 +0x49f
main.write({0x9e51e8, 0xc0000300e8}, {0x9e[31](https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732#step:8:32)d8, 0xc00000e6c0}, 0x186e99319b3)
	/home/runner/work/ceresdb/ceresdb/integration_tests/sdk/go/main.go:[36](https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732#step:8:37) +0x46d
main.main()
	/home/runner/work/ceresdb/ceresdb/integration_tests/sdk/go/main.go:118 +0x1a5
exit status 2
```
<!---
What this problem is and what happened.
-->

**Steps to reproduce**

<!---
How to reproduce this problem.
-->

**Expected behavior**

<!---
What is expected.
-->

**Additional Information**

<!---
If possible, please attach other context you think may relate to this problem. Like *runtime environment*, *modified config*, *error log* etc.
-->

--- a/ceresdb/code.go
+++ b/ceresdb/code.go
@@ -18,12 +18,14 @@ var (
 	ErrNullRequestTables   = errors.New("null request tables")
 	ErrEmptyRoute          = errors.New("empty route")
 	ErrOnlyArrowSupport    = errors.New("only arrow support now")
+	ErrResponseHeaderMiss  = errors.New("response header miss")
 )
 
 const (
 	codeSuccess      = 200
 	codeInvalidRoute = 302
 	codeShouldRetry  = 310
+	codeInternal     = 500
 	codeFlowControl  = 503
 )
 

--- a/ceresdb/rpc.go
+++ b/ceresdb/rpc.go
@@ -37,8 +37,8 @@ func (c *rpcClient) SQLQuery(ctx context.Context, endpoint string, req SQLQueryR
 	if err != nil {
 		return SQLQueryResponse{}, err
 	}
-	grpcClient := storagepb.NewStorageServiceClient(grpcConn)
 
+	grpcClient := storagepb.NewStorageServiceClient(grpcConn)
 	queryRequest := &storagepb.SqlQueryRequest{
 		Context: &storagepb.RequestContext{
 			Database: req.ReqCtx.Database,
@@ -50,7 +50,8 @@ func (c *rpcClient) SQLQuery(ctx context.Context, endpoint string, req SQLQueryR
 	if err != nil {
 		return SQLQueryResponse{}, err
 	}
-	if queryResponse.Header.Code != codeSuccess {
+
+	if queryResponse.Header != nil && queryResponse.Header.Code != codeSuccess {
 		return SQLQueryResponse{}, &CeresdbError{
 			Code: queryResponse.Header.Code,
 			Err:  queryResponse.Header.Error,
@@ -68,6 +69,7 @@ func (c *rpcClient) SQLQuery(ctx context.Context, endpoint string, req SQLQueryR
 	if err != nil {
 		return SQLQueryResponse{}, err
 	}
+
 	return SQLQueryResponse{
 		SQL:          req.SQL,
 		AffectedRows: queryResponse.GetAffectedRows(),
@@ -80,12 +82,13 @@ func (c *rpcClient) Write(ctx context.Context, endpoint string, reqCtx RequestCo
 	if err != nil {
 		return WriteResponse{}, err
 	}
-	grpcClient := storagepb.NewStorageServiceClient(grpcConn)
 
+	grpcClient := storagepb.NewStorageServiceClient(grpcConn)
 	writeRequest, err := buildPbWriteRequest(points)
 	if err != nil {
 		return WriteResponse{}, err
 	}
+
 	writeRequest.Context = &storagepb.RequestContext{
 		Database: reqCtx.Database,
 	}
@@ -93,12 +96,14 @@ func (c *rpcClient) Write(ctx context.Context, endpoint string, reqCtx RequestCo
 	if err != nil {
 		return WriteResponse{}, err
 	}
-	if writeResponse.Header.Code != codeSuccess {
+
+	if writeResponse.Header != nil && writeResponse.Header.Code != codeSuccess {
 		return WriteResponse{}, &CeresdbError{
 			Code: writeResponse.Header.Code,
 			Err:  writeResponse.Header.Error,
 		}
 	}
+
 	return WriteResponse{
 		Success: writeResponse.Success,
 		Failed:  writeResponse.Failed,
@@ -122,7 +127,8 @@ func (c *rpcClient) Route(endpoint string, reqCtx RequestContext, tables []strin
 	if err != nil {
 		return nil, err
 	}
-	if routeResponse.Header.Code != codeSuccess {
+
+	if routeResponse.Header != nil && routeResponse.Header.Code != codeSuccess {
 		return nil, &CeresdbError{
 			Code: routeResponse.Header.Code,
 			Err:  routeResponse.Header.Error,
@@ -131,10 +137,9 @@ func (c *rpcClient) Route(endpoint string, reqCtx RequestContext, tables []strin
 
 	routes := make(map[string]route, len(routeResponse.Routes))
 	for _, r := range routeResponse.Routes {
-		endpoint := fmt.Sprintf("%s:%d", r.Endpoint.Ip, r.Endpoint.Port)
 		routes[r.Table] = route{
 			Table:    r.Table,
-			Endpoint: endpoint,
+			Endpoint: fmt.Sprintf("%s:%d", r.Endpoint.Ip, r.Endpoint.Port),
 		}
 	}
 	return routes, nil
@@ -162,6 +167,7 @@ func (c *rpcClient) newGrpcConn(endpoint string) (*grpc.ClientConn, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	c.connPool.Store(endpoint, conn)
 	return conn, nil
 }

--- a/ceresdb/rpc.go
+++ b/ceresdb/rpc.go
@@ -51,7 +51,14 @@ func (c *rpcClient) SQLQuery(ctx context.Context, endpoint string, req SQLQueryR
 		return SQLQueryResponse{}, err
 	}
 
-	if queryResponse.Header != nil && queryResponse.Header.Code != codeSuccess {
+	if queryResponse.Header != nil {
+		return SQLQueryResponse{}, &CeresdbError{
+			Code: codeInternal,
+			Err:  ErrResponseHeaderMiss.Error(),
+		}
+	}
+
+	if queryResponse.Header.Code != codeSuccess {
 		return SQLQueryResponse{}, &CeresdbError{
 			Code: queryResponse.Header.Code,
 			Err:  queryResponse.Header.Error,
@@ -97,7 +104,14 @@ func (c *rpcClient) Write(ctx context.Context, endpoint string, reqCtx RequestCo
 		return WriteResponse{}, err
 	}
 
-	if writeResponse.Header != nil && writeResponse.Header.Code != codeSuccess {
+	if writeResponse.Header != nil {
+		return WriteResponse{}, &CeresdbError{
+			Code: codeInternal,
+			Err:  ErrResponseHeaderMiss.Error(),
+		}
+	}
+
+	if writeResponse.Header.Code != codeSuccess {
 		return WriteResponse{}, &CeresdbError{
 			Code: writeResponse.Header.Code,
 			Err:  writeResponse.Header.Error,
@@ -128,7 +142,14 @@ func (c *rpcClient) Route(endpoint string, reqCtx RequestContext, tables []strin
 		return nil, err
 	}
 
-	if routeResponse.Header != nil && routeResponse.Header.Code != codeSuccess {
+	if routeResponse.Header != nil {
+		return nil, &CeresdbError{
+			Code: codeInternal,
+			Err:  ErrResponseHeaderMiss.Error(),
+		}
+	}
+
+	if routeResponse.Header.Code != codeSuccess {
 		return nil, &CeresdbError{
 			Code: routeResponse.Header.Code,
 			Err:  routeResponse.Header.Error,
@@ -561,7 +582,6 @@ func convertArrowRecordToRow(schema *arrow.Schema, record array.Record) []Row {
 				}
 			}
 		default:
-			//
 		}
 	}
 

--- a/ceresdb/rpc.go
+++ b/ceresdb/rpc.go
@@ -51,7 +51,7 @@ func (c *rpcClient) SQLQuery(ctx context.Context, endpoint string, req SQLQueryR
 		return SQLQueryResponse{}, err
 	}
 
-	if queryResponse.Header != nil {
+	if queryResponse.Header == nil {
 		return SQLQueryResponse{}, &CeresdbError{
 			Code: codeInternal,
 			Err:  ErrResponseHeaderMiss.Error(),
@@ -104,7 +104,7 @@ func (c *rpcClient) Write(ctx context.Context, endpoint string, reqCtx RequestCo
 		return WriteResponse{}, err
 	}
 
-	if writeResponse.Header != nil {
+	if writeResponse.Header == nil {
 		return WriteResponse{}, &CeresdbError{
 			Code: codeInternal,
 			Err:  ErrResponseHeaderMiss.Error(),
@@ -142,7 +142,7 @@ func (c *rpcClient) Route(endpoint string, reqCtx RequestContext, tables []strin
 		return nil, err
 	}
 
-	if routeResponse.Header != nil {
+	if routeResponse.Header == nil {
 		return nil, &CeresdbError{
 			Code: codeInternal,
 			Err:  ErrResponseHeaderMiss.Error(),

